core/state/statedb_arbitrum.go

```diff
@@ -39,31 +39,36 @@ var (
 	stylusEOFMagic       = byte(0xEF)
 	stylusEOFMagicSuffix = byte(0xF0)
 	stylusEOFVersion     = byte(0x00)
+	stylusEOFDictionary  = byte(0x00) // varies with dictionary selection
 
-	StylusPrefix = []byte{stylusEOFMagic, stylusEOFMagicSuffix, stylusEOFVersion}
+	StylusDiscriminant = []byte{stylusEOFMagic, stylusEOFMagicSuffix, stylusEOFVersion}
 )
 
 type ActivatedWasm struct {
 	Asm    []byte
 	Module []byte
 }
 
-// IsStylusProgram checks if a specified bytecode is a user-submitted WASM program.
-// Stylus differentiates WASMs from EVM bytecode via the prefix 0xEFF000 which will safely fail
-// to pass through EVM-bytecode EOF validation rules.
+// checks if a valid Stylus prefix is present
 func IsStylusProgram(b []byte) bool {
-	if len(b) < len(StylusPrefix) {
+	if len(b) < len(StylusDiscriminant)+1 {
 		return false
 	}
-	return bytes.Equal(b[:3], StylusPrefix)
+	return bytes.Equal(b[:3], StylusDiscriminant)
 }
 
-// StripStylusPrefix if the specified input is a stylus program.
-func StripStylusPrefix(b []byte) ([]byte, error) {
+// checks if the specified input is a stylus program.
```

Comment:
This comment seems to be wrong or incomplete. Maybe "If the input is a stylus program, return it with its prefix stripped as well as its version number." (Or dictionary number, or whatever.)